plugins/search-react/src/components/SearchBar/SearchBar.tsx

```diff
@@ -160,30 +162,53 @@ export const SearchBarBase = ({
  */
 export type SearchBarProps = Partial<SearchBarBaseProps>;
 
+const withContext = (Component: ComponentType<SearchBarProps>) => {
+  return forwardRef<unknown, SearchBarProps>((props, ref) => (
+    <SearchContextProvider parentContext>
+      <Component {...props} ref={ref} />
+    </SearchContextProvider>
+  ));
+};
+
 /**
  * Recommended search bar when you use the Search Provider or Search Context.
  *
  * @public
  */
-export const SearchBar = ({ onChange, ...props }: SearchBarProps) => {
-  const { term, setTerm } = useSearch();
-
-  const handleChange = useCallback(
-    (newValue: string) => {
-      if (onChange) {
-        onChange(newValue);
-      } else {
-        setTerm(newValue);
-      }
+export const SearchBar = withContext(
+  forwardRef<unknown, SearchBarProps>(
+    ({ value: initialValue = '', onChange, ...rest }, ref) => {
+      const { term, setTerm } = useSearch();
```

Comment:
can we destruct the components props here instead? I think that will improve the readability of the api reports a bit.